python/test/unit/language/test_core_amd.py

```diff
@@ -1112,15 +1116,15 @@ def matmul_kernel(
         a_ptrs = a_ptr + (offs_am[:, None] * stride_am + offs_k[None, :] * stride_ak)
         b_ptrs = b_ptr + (offs_k[:, None] * stride_bk + offs_bn[None, :] * stride_bn)
 
-        accumulator = tl.zeros((BLOCK_SIZE_M, BLOCK_SIZE_N), dtype=compute_type)
+        accumulator = tl.zeros((BLOCK_SIZE_M, BLOCK_SIZE_N), dtype=tl.float32)
         for k in range(0, tl.cdiv(K, BLOCK_SIZE_K)):
             a = tl.load(a_ptrs, mask=offs_k[None, :] < K - k * BLOCK_SIZE_K, other=0.0)
             b = tl.load(b_ptrs, mask=offs_k[:, None] < K - k * BLOCK_SIZE_K, other=0.0)
             # We accumulate along the K dimension.
-            accumulator += tl.dot(a, b, out_dtype=compute_type)
+            accumulator += tl.dot(a, b)
```

Comment:
Is there a particular reason to remove the "out_dtype=compute_type" here?

Reply:
mfma has an output type fp32, if we put a type, e.g. fp16, here, the conversion from fp32 to fp16 happens for every iteration of k, which can lose output precision and cause additional overhead. So, I just remove the conversion here and accumulate in fp32, then do the conversion only for the final output.